src/main.cpp

```diff
@@ -2835,8 +2815,7 @@ namespace Consensus {
                 // Disabled on regtest
                 if (fCoinbaseEnforcedProtectionEnabled &&
                     consensusParams.fCoinbaseMustBeProtected &&
-                    !(tx.vout.size() == 0 || (tx.vout.size() == 1 && tx.vout[0].nValue == 0)) &&
-                    (strcmp(ASSETCHAINS_SYMBOL, "VRSC") != 0 || (nSpendHeight >= 12800 && coins->nHeight >= 12800))) {
+                    !tx.vout.empty()) {
```

Comment:
Initially this condition was to prevent spending coinbase as transparent coins (enfore coinbase protection, mean allow to spend coinbase only as shielded coins). The condition `tx.vout.size() == 1 && tx.vout[0].nValue == 0` expand the rule to treat spend coinbase to transparent output with `value = 0` as valid also. Don't sure where we need to use it, from my point of view such condition is unnecessarily, bcz it's at least illogically to have zero transparent output during trying coinbase spend.

Anyway, we have:
1. `fCoinbaseEnforcedProtectionEnabled = true`
2. `consensus.fCoinbaseMustBeProtected = false`
In mainnet. So, this condition even don't ever check, bcz we allowed spend coinbase to transparent outputs.